versioned_docs/version-5/actors.mdx

```diff
@@ -49,58 +86,290 @@ An example of the difference between invoking and spawning actors could occur in
 
 ## Actor snapshots
 
-- `actor.getSnapshot()`
+When an actor receives an event, its internal state may change. An actor may emit a **snapshot** when a state transition occurs. You can read an actor's snapshot synchronously via `actor.getSnapshot()`. An actor's snapshot is not necessarily the same as its internal state; rather, it is a value that it wants to _share_ with subscribers. For example, a promise actor has an internal state consisting of `data`, `status`, and other internal properties, but only the `data` is emitted as its snapshot.
+
+```ts
+import { fromPromise, interpret } from 'xstate';
+
+const countLogic = fromPromise(async () => {
+  const count = await fetchCount();
+
+  return count;
+});
+
+const countActor = interpret(countLogic);
+countActor.start();
+
+countActor.getSnapshot(); // logs undefined
+
+// After the promise resolves...
+countActor.getSnapshot(); // logs 42
+```
 
 ## Subscribing to actors
 
-- Subscribes to emitted actor snapshots
+You can subscribe to an actor's snapshot values via `actor.subscribe(observer)`. The observer will receive the actor's snapshot value when it is emitted. The observer can be:
+
+- A plain function that receives the latest snapshot
+- Or an observer object whose `.next(snapshot)` method receives the latest snapshot
 
-- `actor.subscribe(observer)`
-- Plain function `(state) => { ... }`
-- Observer `({ next, error, complete }) => { ... }`
-- `subscription.unsubscribe()` for cleanup
+```ts
+// Observer as a plain function
+const subscription = actor.subscribe((snapshot) => {
+  console.log(snapshot);
+});
+```
+
+```ts
+// Observer as an object
+const subscription = actor.subscribe({
+  next(snapshot) {
+    console.log(snapshot);
+  },
+  error(data) {
+    // ...
+  },
+  complete() {
+    // ...
+  },
+});
+```
+
+The return value of `actor.subscribe(observer)` is a subscription object that has an `.unsubscribe()` method. You can call `subscription.unsubscribe()` to unsubscribe the observer:
+
+```ts
+const subscription = actor.subscribe((snapshot) => {
+  /* ... */
+});
+
+// Unsubscribe the observer
+subscription.unsubscribe();
+```
+
+When the actor is stopped, all observers will automatically be unsubscribed.
 
 ## Initializing actor state
 
-- `interpret(machine, { state }).start()`
-- [Persistence](persistence.mdx)
+You can initialize actor logic at a specific persisted internal state by passing the state in the 2nd `options` argument of `interpret(logic, options)`. This will create an actor that will be started at that persisted state, if the state is compatible with the actor logic:
```

Comment:
```suggestion
You can initialize actor logic at a specific persisted internal state by passing the state in the second `options` argument of `interpret(logic, options)`. If the state is compatible with the actor logic, this will create an actor that will be started at that persisted state:
```